### PR TITLE
[NCL-7076] - remove old BPM invocation code.

### PR DESCRIPTION
**Pull request by matejonnet**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java [hunk 1]
```diff
@@ -97,8 +93,8 @@
     @Inject
     private Notifier notifier;
 
-    @Inject
-    private BpmManager bpmManager;
+    @Deprecated // use stateless approach
+    private AtomicInteger nextTaskId = new AtomicInteger(1);
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java:97]: I don't think we need it. TaskIds were the way how to identify tasks in memory and Maitai AFAIK.

File: facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java [hunk 2]
```diff
@@ -339,51 +338,24 @@ private Integer startRCreationTask(
...
-            }
+        RepositoryCreationProcess.RepositoryCreationProcessBuilder repositoryCreationProcess = RepositoryCreationProcess
+                .builder()
+                .repositoryConfiguration(repositoryConfiguration)
+                .revision(revision);
+
+        buildConfiguration.ifPresent(bc -> repositoryCreationProcess.buildConfiguration(bc));
+        task = new RepositoryCreationTask(repositoryCreationProcess.build(), userToken);
+
+        task.setTaskId(nextTaskId.getAndIncrement()); // TODO why do we need this ID
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java:349]: I don't think it's needed. Is it used in RHPAM?

**Review by michalovjan — CHANGES_REQUESTED** [review]

File: bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java [hunk 3]
```diff
@@ -54,51 +42,22 @@ public abstract class BpmTask implements Comparable<BpmTask> {
 
     private static final Logger log = LoggerFactory.getLogger(BpmTask.class);
 
-    protected static final ObjectMapper MAPPER = new ObjectMapper();
-
     /**
      * This is an internal identifier, not one provided by BPM server.
      */
     private Integer taskId;
```
[the remark this answers is not in the document]
Comment (matejonnet) [reply bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java:48]: Yes, only in one case, in the notifications of Repository Creation. I wanted to remove it but it's out of the scope for this task.

File: bpm/src/main/java/org/jboss/pnc/bpm/ConnectorFactory.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
+ */
+@ApplicationScoped
+public class ConnectorFactory {
```
[the remark this answers is not in the document]
Comment (matejonnet) [reply bpm/src/main/java/org/jboss/pnc/bpm/ConnectorFactory.java]: Removed the factory.

File: bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (michalovjan) [reply bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java:48]: Can we create a JIRA ticket for it then?

**Review by michalovjan — CHANGES_REQUESTED** [review]
Conflict, otherwise I LIKE IT!

File: bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (matejonnet) [reply bpm/src/main/java/org/jboss/pnc/bpm/BpmTask.java:48]: NCL-7407

**Pull request merged**